### PR TITLE
Fix Stroustrup record member declaration indent issue

### LINKED ISSUE #2787: Record member declarations can break with Stroustrup enabled
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKA7AhgW0QZwAdNpEwBlWXAOgDkd8iSAdAJyfVfYG0AeASXQAXRCwBmxRAD4AupyEBPAqT4BRAG6IhYALxgAlkJHiSYTfDlzBi0hVzrNgnXIgRgYPvDR5BLAwHNnFwg6XC8ffzAAX0CIA2ExCXd7IQs2BSUwAFUCeExheBDSXTcABRZENT1YAFc8QrDfdD8o1PZIABpwKDxEQUF-PFQwLi7XUZdIABIrJSHIOI7xiCm1TAAbauQ0EaCggAZO3YmBJCEyPQAvZEOjsAAWJbBpccibsaOpma2oBbeJydWG2%2BO1uAEY-kFIABZTAADwAMgZEPDNH5BAALRa3MCggBMe0ezyCr3GoA%2B02sczMAHlRIj0IgyFY1tdHit1pshiCjriIRMVOh4LT6azsZA1qJIISXn8ybtPpS0JAAEawWBrLHygEc4GPCAAZj5ywEPRYggAYgZ1jREAB3NZIzVHHybaXE2VsimzJWq9VO-6AznbPX3I1QMiMRAAIUQolg5RKmBY9Hi-qCLsQbpcJKCcshXu%2BKrVGr57KBXJDAFYw5AIxIY3HyvDYLaRNBMD0BGpYO3%2BrB0GmXBmsxAcy48-8vnNfSXPYHddiAGw1uskBvxxDZJQsdud9Dd3tVAdh8RrHojqIe8lTn3FwdloPDEMAdhXkfX5QAwmsO3hP-3vBYapoEEeNBwgU9zyOIlsyvLUbygGd721ctg2xAAON961jDcoUQbBlREcCwEgzNoJlUlPQQos-VLFDH25XYAE4sLXHCv3VftiNIi8x3eeDFUQu86PnCtsVBA4Q1rSMAEFRHif9sGwTBiOHcj3Uo69BJo2dyVEtCwXBKTV2jdjGXwvQ4DWLiT3WKDdhg0c4PzaikJEnUxLBXljNk%2BSRDICyrJskM1Icijcyo7S3LnDyDKOUFDR8iQZJYGpBQAEUQB1sD0VMw1CoJHMvTSBO9H4hGQ-Sn3Eu4axhWE%2BFEAAVdFNDIdF40EAB1PR4AxYiCXU2CSpc7Tfhi1DqrBaspPqxqWs0FQz0ZDqzR6vrMTDRdBrCjSIq0sr5gq9zJsYoJQWXWa4QEUQ9FhaltzyeMVFhAhyjwPAj2I9CdsK8Lx0iw7xr02Kpvi18rthAAlRA4BYeB1v6sM7l%2Blwir4sAJ2WAs5mBrUqrOlxQUwyGYbhgpqgIkRaT4YRsEGMNQV45zJ207BqjWfoHQZc142UvoRCaykToYkNQRYqSyfjeAoQ5rmkV5lh%2BbyqToHRJNiHiAB9W1ev65mRtZoHjom0XsXxOq4RS5N5GpFhEW8RHNpDH6Df20rCzx-MCZDXEjLFerrcwW37b0bwaEpwiWBpumGbFt2AYOwt2c5vRucQRXlcF4XTYXW5cW8sUg5Dh3BFl1P08zvIVbFNWNZAkQdb1zEE-40bjcESrQcJiBcUSgO4QANR1KMDHgfwne%2B1GIHRlnseor2A2733ash81qnQECj1HwUJ%2Bb4iUdbzHAc9k2QdO32ZoH2F0tgQQAHFehet78E%2B-tJ7DV2hqcw356i4Tc6eR5JdMUNp7RIhjIIVsmghZKEyrddAuUjwyUFHhKmLA47YgKmjf6bcjaFmiufM2%2BcIZFwdH4dA69N59nQOcCheRqjlCarAE4Dg8hfVsstI%2BWMoA41vLRQBcVdi4hJmQ%2BIWB%2BgaGbE0NBUd4EGCQQBbidkyK7WGu7duBCAFELzjyCWAc5Z6BFPCHAypcjflgJ9JoYD07KK4d-YqGj8HTm0fjZe2J9SSTFM-EQehcBCHWAAaUQIgAgrChACCjMmTeztsQ8QcRjHhCpDqELcRfDx-tbiQCjD%2BdAABrEUeAUppQKPgfI5d5YMmfu9N%2B6BMG3GwTPXBx8k4uIETooBux9SFyyVGJMH5EDpTDtAXwOUJGIHgJkRB-ZMrQB-MmGhdj7J-T2onD2cwU6VOjMmaAeTehMnkCyLu6Tbj6n7lkipacIE7L2YIA5RyayARqIBaoBApQJLnrwheZ80nEKOPqVeYpgmhPqhHdBtIcmYHyYU4iElp5PGaUkvhQl2m-N0V0q%2BWSbEQLMpc9OikCDVEEOw-s1TX4cJCkBVRKz1FrM0W03SaLOlBH1CArJTJfAgShAgUUtx4lqNHF0aQNx5h4HNJ9IYPFIhAA)

#### Code

```fsharp
namespace Some.Namespace

[<Interface>]
type IEvent = interface end

type SomeEvent =
    { Id: string
      Name: string }
    interface IEvent

type UpdatedName = { PreviousName: string }
```
            

#### Result

```fsharp
namespace Some.Namespace

[<Interface>]
type IEvent =
    interface
    end

type SomeEvent = {
    Id: string
    Name: string
} with

    interface IEvent

    type UpdatedName = { PreviousName: string }

```
            
#### Problem description

We use an empty interface for a few things, and updating my code to run the latest alpha seems to break in those places when I turn on the Stroustrup setting if anything comes afterwards. I'll try and get a PR up for this soon.

#### Extra information

- [x] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas v6 branch at 3/6/23

```fsharp
    { config with
                MultilineBracketStyle = Stroustrup }
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>

### PR DESCRIPTION
**Pull request by josh-degraw**
Closes #2787

### CODE REVIEW

File: src/Fantomas.Core/CodePrinter.fs [hunk 1]
```diff
@@ -3234,9 +3234,12 @@ let genTypeDefn (td: TypeDefn) =
...
             let genMembers =
-                onlyIf hasMembers sepNln
-                +> sepNlnBetweenTypeAndMembers typeDefnNode
-                +> genMemberDefnList members
+                onlyIf
+                    hasMembers
+                    (sepNln
+                     +> sepNlnBetweenTypeAndMembers typeDefnNode
+                     +> genMemberDefnList members
+                     +> onlyIfCtx (fun ctx -> ctx.Config.IsStroustrupStyle) unindent)
```
Comment (nojaf) [inline src/Fantomas.Core/CodePrinter.fs:3242]: A bit of a nit but would this not make more sense to have it closer to the `indent`?
Something like:
```
                genAccessOpt node.Accessibility
                +> genRecordFields
                +> onlyIf hasMembers (sepSpace +> withKw +> indent +> genMembers +> unindent)
```

Comment (josh-degraw) [reply src/Fantomas.Core/CodePrinter.fs]: Good call.

**Comment by josh-degraw** [comment]
@nojaf Would you mind if we made a new alpha release with the latest changes? It looks like this change seems to solve all the issues I found locally and I might be able to actually get my team's code updated to use Stroustrup again finally with this change haha

**Comment by nojaf** [comment]
Yup, just add the release notes to the v6 branch and publish from there.
No need for a github release. We will do that eventually.

**Pull request merged**